test/interval.jl

```diff
@@ -180,6 +206,36 @@
             end
         end
 
+        ###################
+        # Unbounded tests #
+        ###################
+        # Disjoint
+        @test !(Interval(-∞, ∞) ≪ Interval(-∞, ∞))
+        @test !(Interval(-∞, ∞) ≪ Interval(-100, 100))
+        @test !(Interval(-∞, ∞) ≫ Interval(-∞, ∞))
+        @test !(Interval(-∞, ∞) ≫ Interval(-100, 100))
+        @test Interval(-100, 100) ≪ Interval(200, ∞)
+        @test !(Interval(-100, 100) ≪ Interval(50, ∞))
+        @test Interval(-100, 100) ≫ Interval(-∞, -200)
+        @test !(Interval(-100, 100) ≫ Interval(-∞, -50))
+
+        # Less/Greater than
+        @test !isless(Interval(-∞, ∞), Interval(-∞, ∞))
+        @test !(Interval(-∞, ∞) < Interval(-∞, ∞))
+        @test !(Interval(100, 200) < Interval(-∞, ∞))
+        @test Interval(100, 200) < Interval(150, ∞)
+        @test !(Interval(-∞, ∞) > Interval(-∞, ∞))
+        @test Interval(100, 200) > Interval(-∞, ∞)
+        @test Interval(100, 200) > Interval(-∞, 150)
+        @test Interval(150, 200) > Interval(100, ∞)
+
+        # Equality
+        @test Interval(-∞, ∞) == Interval(-∞, ∞)
+        @test Interval(-∞, ∞) != Interval(-∞, ∞, true, false)
+        @test Interval(-∞, ∞) != Interval(-∞, ∞, false, true)
+        @test Interval(-∞, ∞) != Interval(-∞, ∞, false, false)
```

Comment:
I thought we weren't allowing inclusive endpoints for unbounded endpoints?

Reply:
I wasn’t sure if we still wanted to do that with proper infinity. As it potentially makes sense for infinity itself to be inclusive or not? It definitely makes the logic nicer to include inclusivity. I am willing to go back to mutating the inclusivity.

Reply:
As infinity is the first non-finite value we're supporting in intervals I think there is going to be some specialized logic around it. I think I'll write some of the logic down in the original issue.

Reply:
After writing down specific rules I think we can leave this as it is: https://github.com/invenia/Intervals.jl/issues/89#issuecomment-635372868